src/httpClient.ts

```diff
@@ -1,7 +1,11 @@
-import type { Response } from "undici";
+import type { Dispatcher } from "undici";
 import { fetch, Agent } from "undici";
 import pako from "pako";
 import { version } from "../package.json";
+import { gunzip } from "node:zlib";
+import { promisify } from "node:util";
+
+const gunzip_async = promisify(gunzip);
```

Comment:
Why is this required? Presumably it's still blocking underneath, so what's the reason to not just call `gunzip`?

Reply:
it's callback based, not blocking, promisify makes it possible to await it

gunzipSync is blocking and would be really bad to use on a production server